packages/devtools_app/lib/src/screens/memory/shared/heap/model.dart

```diff
@@ -44,16 +43,13 @@ class AdaptedHeapData {
     );
   }
 
-  static Future<AdaptedHeapData> fromHeapSnapshot(
+  static AdaptedHeapData fromHeapSnapshot(
     HeapSnapshotGraph graph,
-  ) async {
-    final listOfFutures = graph.objects.map((e) async {
-      await delayForBatchProcessing();
```

Comment:
I think this might have been causing some of your issues. Can you try not awaiting this and see if that improves performance at all? Something like this:

```dart
final listOfFutures = graph.objects.map((e) {
  return delayForBatchProcessing().then(
    (_) async => await AdaptedHeapData.fromHeapSnapshot(graph),
  );
});
await Future.wait(listOfFutures);
```

Reply:
tried. did not help :(

Reply:
I'm fairly certain this is because we are awaiting a delay for every object in the graph. We should not do this. See my comment here: https://github.com/flutter/devtools/pull/4676#discussion_r1010542209